### PR TITLE
No need to use bot IDS, use of the readable label instead

### PR DESCRIPTION
**Pull request by MrWaloo**
The strategies information can be found at https://middle.napbots.com/v1/strategy. It is in json format and contains the corresponding ID for each bot.
The new function get_strategies_code() parses the json file.
The function assign_composition() as  een adapted to 'translate' the configured compositions to coded compositions (aka with IDS). In this function "global $debug;" was missing.
Small code correction BTW.

### CODE REVIEW

File: manage-allocations.php [hunk 1]
```diff
@@ -73,17 +72,47 @@ function get_market(){
...
+	$strategiesApi = file_get_contents('https://middle.napbots.com/v1/strategy');
+	$codes = [];
+	if($strategiesApi) {
+		$strategies = json_decode($strategiesApi, true)['data'];
+		foreach($strategies as $strat){
+			$codes[$strat['label']] = $strat['code'];
+		}
+	}
+	return $codes;
+}
```
Comment (MrWaloo) [inline manage-allocations.php:86]: Missing some debug...

File: manage-allocations.php [hunk 2]
```diff
@@ -73,17 +72,47 @@ function get_market(){
...
+				$val = [];
+				foreach($value as $label => $percentage) {
+					$val[$strategies_code[$label]] = $percentage;
+				}
+				$coded_compositions[$comp_weather][$config] = $val;
+			} else {
+				$coded_compositions[$comp_weather][$config] = $value;
+			}
+		}
+	}
```
Comment (MrWaloo) [inline manage-allocations.php:107]: Missing some debug
The new line #109 """if($debug) echo "enter assign_composition [$weather]\n";""" should be on the top of this function

**Review by MrWaloo — COMMENTED** [review]
Comments for me...

**Comment by julienarcin** [comment]
Thanks for the PR.

But I am not ok to change the bots ids to readable labels in config files for 3 reasons : 
 1. Using names in a config file is a bad practice in general. Names can change, as it had a few weeks ago (Napbots Hourly 2 became Napbots Ultra Flex).
 2. This is more prone to errors, as a missing slash or space will cause problems.
 3. It is not backward compatible. The users who are already using the script will have to update the config file with bot names.

I understand that using bot ids is less user friendly, but I think it's better to work on the documentation than implementing a less viable solution in the code.

**Comment by julienarcin** [comment]
By the way I like your solution for the crypto weather problem, could you make a separate PR ?

Thanks.

**Comment by MrWaloo** [comment]
Hi Julien,

> Thanks for the PR.
> 
> But I am not ok to change the bots ids to readable labels in config files for 3 reasons :
> 
> 1. Using names in a config file is a bad practice in general. Names can change, as it had a few weeks ago (Napbots Hourly 2 became Napbots Ultra Flex).
As you can see, the names are not defined in the code but come from the napbots site
> 2. This is more prone to errors, as a missing slash or space will cause problems.
Since the names come from the site, they cannot contain errors. But... (see comment at the end of this message)
> 3. It is not backward compatible. The users who are already using the script will have to update the config file with bot names.
To change the config to an easier one just make sense...
> 
> I understand that using bot ids is less user friendly, but I think it's better to work on the documentation than implementing a less viable solution in the code.

For this point I think to make a better check of the configuration and alert the user in case of an error in the name.
But if you don't want to implement this solution, no worry, the principle of a fork will permit the users to make the choice between the original version (yours) and mine.
Really don't worry :-)

**Comment by MrWaloo** [comment]
> By the way I like your solution for the crypto weather problem, could you make a separate PR ?
> 
> Thanks.

OK, I will give it a try

**Pull request closed without merging**